### PR TITLE
Crop rotate and emojis

### PR DESCRIPTION
**Pull request by Bwananah**
PR for 3 photo editor features :
- Image cropping
- Emoji stickers
- Rotating the image using a 2 finger gesture (AltGr + mouse on the emulator)

### CODE REVIEW

File: app/src/main/java/com/github/epfl/meili/photo/PhotoCropActivity.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        }
+        return Bitmap.createBitmap(original, 0, 0, original.width, original.height, matrix, true)
+    }
+
+    /** Launch PhotoEditActivity */
+    private fun launchEffects() {
+        // Save rotated image into uri
+        try {
+            FileOutputStream(uri.path).use { out ->
+                getRotatedBitmap().compress(Bitmap.CompressFormat.PNG, 100, out)
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/photo/PhotoCropActivity.kt:116]: Using a named constant for 100 would be clearer

File: app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt [hunk 2]
```diff
@@ -154,11 +161,46 @@ class PhotoEditActivity : AppCompatActivity() {
...
 
+    /** Creates all textView in the emoji table */
+    private fun makeEmojiTable() {
+        val emojis = PhotoEditor.getEmojis(this) // list of emojis (emojis are strings)
+        var curRow = TableRow(this)
+
+        // Make a new textView for each emoji and arrange in grid
+        for (i in 0 until emojis.size) {
+            // Make new row every 6 emojis
+            if (i % 6 == 0 && i != 0) {
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt:182]: Same here

File: app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt [hunk 3]
```diff
@@ -154,11 +161,46 @@ class PhotoEditActivity : AppCompatActivity() {
...
+                binding.emojiTable.addView(curRow)
+                curRow = TableRow(this)
+            }
+
+            // Make textView
+            val textView = TextView(this)
+            textView.textSize = 50f
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/photo/PhotoEditActivity.kt:189]: same

**Review by tberkane — APPROVED** [review]
Looks great! Just a few nitpicks

**Review by aman3014 — APPROVED** [review]
The feature looks awesome! Well done for great coverage too!

**Pull request merged**